setup.py

```diff
@@ -34,6 +34,7 @@
     "google-cloud-core >= 1.4.1, < 2.0dev",
     "google-resumable-media >= 0.6.0, < 2.0dev",
     "protobuf >= 3.12.0",
+    "python-dateutil >= 2.8.1",  # used in tests, otherwise dependency of "pandas" extra
```

Comment:
When adding dependencies, could you also add to https://github.com/googleapis/python-bigquery/blob/master/testing/constraints-3.6.txt so that we know the minimum version we specify does actually work?

Reply:
Also, 2.8.1 is the latest version of python-dateutil. Are you sure we can't use a wider range?

Reply:
Also also, do we actually need this dependency? I see it's commonly used and BSD-licensed, so should be OK, but I like to avoid new deps where possible.

If truly test-only, we can put this in the noxfile instead.

Reply:
I used it because the issue reporter's example used its `tzoffset()` helper to specify a timezone offset. But on a second though you are right, adding a dependency just for that is an overkill, even if we currently do implicitly install it as a sub-dependency.

I'll just manually define a custom `datetime.tzinfo` subclass like in some other tests, and we don't even have to bother much about the logic of its `dst()` method, as dateutil's [tzoffset](https://github.com/dateutil/dateutil/blob/24caabfb63c7dcbf6499b3ee5b6140e6b7da0d8a/dateutil/tz/tz.py#L132-L158) simply returns zero.